### PR TITLE
new recipe: lilypond

### PR DESCRIPTION
**Pull request by jmairboeck**
~Tests fail because of missing `extractbb` command (which should be provided by `texlive_core`).~ fixed in 29e1618 in #6818 

It also cannot be built right now as long as `texlive` is not available (should it be marked as untested?). Texlive is only required for building, not at runtime.

The libgs API is used instead of the GhostScript executable because that crashes on Lilypond's PostScript output (see #6284).

### CODE REVIEW

**Comment by jmairboeck** [comment]
It seems that using extractpdfmark works now with the current version of ghostscript (Lilypond itself still doesn't but that is not a big problem). This means that the size of the doc packages is now about half of before. :)

**Comment by jmairboeck** [comment]
Now the tests fail because xelatex doesn't find the DejaVu fonts. I don't know yet how to fix that.

File: media-sound/lilypond/lilypond-2.22.2.recipe [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+	devel:libharfbuzz$secondaryArchSuffix
+	devel:libintl$secondaryArchSuffix
+	devel:libltdl$secondaryArchSuffix
+	devel:libpangoft2_1.0$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+#	cmd:awk
+	cmd:bison
+	cmd:convert$secondaryArchSuffix
+	cmd:dblatex
```
Comment (Begasus) [inline media-sound/lilypond/lilypond-2.22.2.recipe:103]: cmd not found?

**Comment by jmairboeck** [comment]
dblatex is in #6078

File: media-sound/lilypond/lilypond-2.22.2.recipe [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+	cmd:mf
+	cmd:mpost
+	cmd:perl
+	cmd:pkg_config$secondaryArchSuffix
+	cmd:rsync
+	cmd:t1asm
+	cmd:tar
+	cmd:texi2html
+	cmd:texindex
+	cmd:tidy$secondaryArchSuffix
```
Comment (Begasus) [inline media-sound/lilypond/lilypond-2.22.2.recipe:122]: No $secondaryArchSuffix here:
```diff
- 	cmd:tidy$secondaryArchSuffix
+ 	cmd:tidy
```

File: media-sound/lilypond/lilypond-2.22.2.recipe [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+HOMEPAGE="https://lilypond.org/"
+COPYRIGHT="1993-2020 The LilyPond development team"
+LICENSE="GNU GPL v3"
+REVISION="1"
+SOURCE_URI="https://lilypond.org/download/source/v${portVersion%.*}/lilypond-$portVersion.tar.gz"
+CHECKSUM_SHA256="dde90854fa7de1012f4e1304a68617aea9ab322932ec0ce76984f60d26aa23be"
+PATCHES="lilypond-$portVersion.patchset"
+
+ARCHITECTURES="all !x86_gcc2"
+SECONDARY_ARCHITECTURES="x86"
```
Comment (Begasus) [inline media-sound/lilypond/lilypond-2.22.2.recipe:14]: Mark as untested `?`, build is OK but needs texlive and dblatex

**Comment by Begasus** [comment]
Needs texlive PR merged first: https://github.com/haikuports/haikuports/pull/6818, then dblatex PR merged before: https://github.com/haikuports/haikuports/pull/6078

File: media-sound/lilypond/lilypond-2.22.2.recipe [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+	cmd:dblatex
+	cmd:extractpdfmark
+	cmd:fc_list$secondaryArchSuffix
+	cmd:find
+	cmd:flex
+	cmd:fontforge
+	cmd:gcc$secondaryArchSuffix
+	cmd:kpsewhich
+	cmd:make
+	cmd:mawk # required by texindex
```
Comment (jmairboeck) [inline media-sound/lilypond/lilypond-2.22.2.recipe:112]: this is because of #6020

**Comment by jmairboeck** [comment]
with a complete fontconfig cache, the tests run successfully on my system.

File: media-sound/lilypond/lilypond-2.22.2.recipe [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+			$(find $docDir -name "*.${lang}.*")
+	done
+
+	# generic and english doc package
+	packageEntries doc \
+		$docDir $infoDir
+}
+
+TEST()
+{
```
Comment (jmairboeck) [inline media-sound/lilypond/lilypond-2.22.2.recipe:180]: should I add `fc-cache -f` here to make sure?

Comment (Begasus) [reply media-sound/lilypond/lilypond-2.22.2.recipe:178]: And in TEST_REQUIRES also, could you run the tests with/without the urw fonts? (I'm guessing it searches for the gostscript fonts?)

Comment (jmairboeck) [reply media-sound/lilypond/lilypond-2.22.2.recipe:178]: the URW fonts are in BUILD_REQUIRES anyway. I think TEST_REQUIRES just contains the additional requirements for tests?

Comment (Begasus) [reply media-sound/lilypond/lilypond-2.22.2.recipe:178]: Yes, like fc-cache isn't needed at runtime nor build time, so only needs adding in TEST_REQUIRES :) (and as you mentioned it above in TEST() (PS, not goind to build this monster again to check, so I take your word for it) :)

**Pull request merged**